### PR TITLE
Added java test for addresses

### LINKED ISSUE #73: Add tests for verifying correct address derivation from seed
## Description

We lack tests that verify that address derivation is at par across implementations and bindings i.e., rust implementation and (java, nodejs, python) bindings. We should add them to be sure every implementation has correct address derivation.

Related ticket: https://github.com/iotaledger/iota.rs/issues/790 

## Motivation

Extensive tests like these are needed to be confident about deterministic address derivation across all implementations. 

## Requirements

For a set of mnemonics, ensure they lead to deterministic addresses for different indexes. 

- [x] Python — @bingyanglin;
- [ ] Java — @kwek20;
- [x] Rust — @Thoralf-M;
- [x] NodeJS — @laumair;
- [x] Wasm — @laumair 

The mnemonics and expected addresses should be shared between all tests. 

**Tests should cover both derivations from mnemonics and also from hex seeds.**

## Open questions (optional)

Happy to hear if there are any other similar test scenarios that should be covered. 

## Are you planning to do it yourself in a pull request?

Yes — for nodejs bindings.

### PR DESCRIPTION
**Pull request by kwek20**
# Description of change

Adds an address generation test to java

## Links to any relevant issues

fixes https://github.com/iotaledger/iota-sdk/issues/73

### CODE REVIEW

File: wallet/bindings/java/lib/src/test/java/org/iota/WalletTests.java [hunk 1]
```diff
@@ -49,4 +53,32 @@ public void testGetAccounts() throws WalletException {
...
 
+    @Test
+    public void testGenerateAddress() throws WalletException {
+        AddressGenerationOptions.GenerateAddressOptions options = new AddressGenerationOptions.GenerateAddressOptions().withLedgerNanoPrompt(false);
+        AddressGenerationOptions addressOptions = new AddressGenerationOptions().withOptions(options);
+
+        String address = wallet.generateAddress(0, false, 0, options, "rms");
+        assertEquals("rms1qpx0mcrqq7t6up73n4na0zgsuuy4p0767ut0qq67ngctj7pg4tm2ynsuynp", address);
+
+        // generated account at line 77 has first 2 pre-made, so we check against the third
```
Comment (Thoralf-M) [inline wallet/bindings/java/lib/src/test/java/org/iota/WalletTests.java]: line 77 isn't right (anymore?)

Comment (kwek20) [reply wallet/bindings/java/lib/src/test/java/org/iota/WalletTests.java]: youre right, ill change to it just say the account address later

**Review by Thoralf-M — DISMISSED** [review]

File: wallet/bindings/java/lib/src/main/java/org/iota/Wallet.java [hunk 2]
```diff
@@ -300,6 +302,23 @@ public void emitTestEvent(WalletEvent event) throws WalletException {
         callBaseApi(new WalletCommand("emitTestEvent", o));
     }
 
+    /**
+     * Generate addresses.
```
Comment (thibault-martinez) [inline wallet/bindings/java/lib/src/main/java/org/iota/Wallet.java]: Singular ?

File: wallet/bindings/java/lib/src/main/java/org/iota/Wallet.java [hunk 3]
```diff
@@ -300,6 +302,23 @@ public void emitTestEvent(WalletEvent event) throws WalletException {
...
+     *
+     * @param options The options.
+     * @return The generated addresses.
```
Comment (thibault-martinez) [inline wallet/bindings/java/lib/src/main/java/org/iota/Wallet.java]: Singular ?

**Review by Thoralf-M — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**